commit abaab49630e882fa6418bbf959a178ff7d86a437 (refs/changes/51/951/1)
Author: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date:   2016-11-24T19:04:09+03:00 (2 years, 10 months ago)
    
    add application key for metadata update in case it does not exist

diff --git a/lib/index.js b/lib/index.js
index b4cfd2d..7904ddd 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2292,6 +2292,9 @@ function modifyMetadata(user, key, md, account, cb) {
 
         self.getMetadata(user, key, account, function (err) {
             if (err && err.statusCode === 404) {
+                if (!md.cn) {
+                    md.cn = key;
+                }
                 self.add(dn, md, function (err2) {
                     if (err2) {
                         cb(translateError(err2));
